crates/ruff_python_semantic/src/model.rs

```diff
@@ -177,7 +176,7 @@ impl<'a> SemanticModel<'a> {
     pub fn match_typing_call_path(&self, call_path: &CallPath, target: &str) -> bool {
         if matches!(
             call_path.as_slice(),
-            ["typing" | "_typeshed" | "typing_extensions", target]
+            ["typing" | "_typeshed" | "typing_extensions", _target]
         ) {
```

Comment:
fyi using `matches!` really broke things because `target` here isn't what you wanted — it's capturing second item in the slice not using the provided `target`.

Reply:
fixed in https://github.com/astral-sh/ruff/pull/8429/commits/545bd30c7bdc50009cf931ad585884cdd2117535 — not sure how else to write it